sdk/tdf_config.go

```diff
@@ -275,6 +277,68 @@ type TDFReaderConfig struct {
 
 	schemaValidationIntensity SchemaValidationIntensity
 	kasSessionKey             ocrypto.KeyPair
+	kasAllowlist              AllowList // KAS URLs that are allowed to be used for reading TDFs
+	ignoreAllowList           bool      // If true, the kasAllowlist will be ignored, and all KAS URLs will be allowed
+}
+
+type AllowList map[string]bool
+
+func getKasAddress(kasURL string) (string, error) {
+	parsedURL, err := url.Parse(kasURL)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse kas url(%s): %w", kasURL, err)
+	}
+
+	// Needed to support buffconn for testing
+	if parsedURL.Host == "" && parsedURL.Port() == "" {
+		return "", nil
+	}
+
+	port := parsedURL.Port()
+	// if port is empty, default to 443.
+	if port == "" {
+		return parsedURL.Hostname(), nil
+	}
+
+	return net.JoinHostPort(parsedURL.Hostname(), port), nil
+}
+
+func newAllowList(kasList []string) (AllowList, error) {
+	allowList := make(AllowList)
+	for _, kasURL := range kasList {
+		err := allowList.Add(kasURL)
+		if err != nil {
+			return nil, fmt.Errorf("error adding kas url(%s) to allowlist: %w", kasURL, err)
+		}
+	}
+	return allowList, nil
+}
+
+func (a AllowList) IsAllowed(kasURL string) bool {
```

Comment:
Can we add a check here that the `kasURL` protocol is `https`, and add a unit test that `ftp` or `http` urls wont be allowed?

Reply:
for all of our opentdf platform instances we use http, a hardcoded check would block testing. but we could possibly add an option to be more strict when making decisions

Reply:
If we can't enforce https at this point I think we should avoid stripping the protocol from the allow list.  And if no protocol is specified, default to https.  Then at this point we can validate the expected protocol matches.

Reply:
@jentfoo Even if someone specified `ftp` our client wouldn't use that protocol. Just trying to understand why matching on the protocol is important in this case.

Reply:
I am not so worried about ftp, it was just another example of possible confusion.  I am more worried about `http` being used when the service should have cert validation.  MitM is possible with http, not just from a clear text perspective, but also from the ability to impersonate the service.

For example the exploit would still be possible if:
* The attacker specified an http kas server that is otherwise on an allowed domain
* The attacker manipulates the host file or performs another DNS attack to now allow the client to talk to the attacker controlled KAS

Reply:
updated to require matching schemes as well